src/main/java/core/basesyntax/models/Fruit.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax.models;
+
+public class Fruit {
+    private Type typeOfFruit;
+
+    public void setTypeOfFruit(Type typeOfFruit) {
+        this.typeOfFruit = typeOfFruit;
+    }
+
+    public Type getTypeOfFruit() {
+        return typeOfFruit;
+    }
+
+    public enum Type {
+        APPLE("apple"),
+        BANANA("banana");
+        private final String code;
+
+        Type(String code) {
+            this.code = code;
+        }
+
+        public String getCode() {
+            return code;
+        }
+
+        public static Type setType(String code) {
+            for (Type type : values()) {
+                if (type.code.equals(code)) {
+                    return type;
+                }
+            }
+            throw new IllegalArgumentException("No enum constant with code: " + code);
+        }
+    }
+}
```

Comment:
you dont use class `Fruit`, so better delete it. you can leave only FruitType enum, but we can avoid use of it too just by using String representations